Mathlib/LinearAlgebra/InvariantBasisNumber.lean

```diff
@@ -302,20 +304,28 @@ private def induced_equiv [Fintype ι'] (I : Ideal R) (e : (ι → R) ≃ₗ[R]
 
 end
 
-section
-
-attribute [local instance] Ideal.Quotient.field
-
-/-- Nontrivial commutative rings have the invariant basis number property.
+/-- Nontrivial commutative semirings satisfy the rank condition.
 
 In fact, any nontrivial commutative ring satisfies the strong rank condition, see
-`commRing_strongRankCondition`. We prove this instance separately to avoid dependency on
-`LinearAlgebra.Charpoly.Basic`. -/
-instance (priority := 100) invariantBasisNumber_of_nontrivial_of_commRing {R : Type u} [CommRing R]
-    [Nontrivial R] : InvariantBasisNumber R :=
-  ⟨fun e =>
-    let ⟨I, _hI⟩ := Ideal.exists_maximal R
-    eq_of_fin_equiv (R ⧸ I)
-      ((Ideal.piQuotEquiv _ _).symm ≪≫ₗ induced_equiv _ e ≪≫ₗ Ideal.piQuotEquiv _ _)⟩
-
-end
+`commRing_strongRankCondition`, but it is not clear whether this generalizes to semirings.
+We prove this instance separately to avoid dependency on `LinearAlgebra.Charpoly.Basic`. -/
+instance (priority := 100) rankCondition_of_nontrivial_of_commSemiring {R : Type*}
+    [CommSemiring R] [Nontrivial R] : RankCondition R where
+  le_of_fin_surjective {n m} f hf := by
+    by_contra! lt
+    let p : (Fin m → R) →ₗ[R] Fin n → R :=
+      ⟨⟨(· ∘ Fin.castLE lt.le), fun _ _ ↦ rfl⟩, fun _ _ ↦ rfl⟩
```

Comment:
Preferable as
```suggestion
      { toFun := (· ∘ Fin.castLE lt.le); map_add' _ _ := rfl; map_smul' _ _ := rfl }
```
as this is invariant to field order

Reply:
Or better
```suggestion
      LinearMap.funLeft R R <| Fin.castLE lt.le
```

Reply:
Great find! I [tried to Loogle it](https://leanprover.zulipchat.com/#narrow/channel/217875-Is-there-code-for-X.3F/topic/.60loogle.60.20queries/near/492571830) but it timed out. I found [LinearMap.compLeft](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/Pi.html#LinearMap.compLeft) (why is it not compRight?) but [LinearMap.compRight](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/LinearMap/End.html#LinearMap.compRight) is not about the Pi module.

Reply:
I found this with `exact?`.

Reply:
Oh, I thought of using it for one second but then forgot 😄